crates/polars-plan/src/logical_plan/builder.rs

```diff
@@ -324,25 +324,25 @@ impl LogicalPlanBuilder {
         truncate_ragged_lines: bool,
         mut n_threads: Option<usize>,
     ) -> PolarsResult<Self> {
-        let path = path.into();
-        let mut file = polars_utils::open_file(&path)?;
-
-        let paths = Arc::new([path]);
-
-        let mut magic_nr = [0u8; 4];
-        let res_len = file.read(&mut magic_nr)?;
-        if res_len < 2 {
-            if raise_if_empty {
-                polars_bail!(NoData: "empty CSV")
+        for path in paths.iter() {
+            let mut file = polars_utils::open_file(path)?;
+
+            let mut magic_nr = [0u8; 4];
+            let res_len = file.read(&mut magic_nr)?;
+            if res_len < 2 {
+                if raise_if_empty {
+                    polars_bail!(NoData: "empty CSV")
+                }
+            } else {
+                polars_ensure!(
+                !is_compressed(&magic_nr),
+                ComputeError: "cannot scan compressed csv; use `read_csv` for compressed data",
+                );
             }
```

Comment:
@ritchie46 do you think that this code should be moved into the csv reader itself (when reading the metadata)?

Reply:
Yes, agree. I am a bit surprised that code is here.